### PR TITLE
libpod: if slirp4netns fails, return its stderr

### PR DESCRIPTION
**Pull request by giuseppe**
read the slirp4netns stderr and propagate it in the error when the
process fails.

Replace: https://github.com/containers/libpod/pull/4338/files

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: libpod/networking_linux.go [hunk 1]
```diff
@@ -210,6 +212,16 @@ func (r *Runtime) setupRootlessNetNS(ctr *Container) (err error) {
...
+	logFile, err := os.Create(logPath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to open slirp4netns log file %s", logPath)
+	}
+	// Unlink immediately the file so we won't need to worry about cleaning it up later.
+	// It is still accessible through the open fd logFile.
+	if err := os.Remove(logPath); err != nil {
+		return errors.Wrapf(err, "delete file %s", logPath)
+	}
+	cmd.Stderr = logFile
```
Comment (AkihiroSuda) [inline libpod/networking_linux.go]: stdout should be captured as well

Comment (giuseppe) [reply libpod/networking_linux.go]: fixed

Comment (AkihiroSuda) [reply libpod/networking_linux.go]: cmd.Stdout seems unset

**Comment by TomSweeneyRedHat** [comment]
Other than the stdout question, LGTM

**Comment by mheon** [comment]
Where do we close the FD for the log file?

File: libpod/networking_linux.go [hunk 2]
```diff
@@ -238,7 +252,15 @@ func (r *Runtime) setupRootlessNetNS(ctr *Container) (err error) {
...
 				}
 				if status.Exited() {
-					return errors.New("slirp4netns failed")
+					// Seek at the beginning of the file and read all its content
+					if _, err := logFile.Seek(0, 0); err != nil {
+						logrus.Errorf("could not seek log file: %q", err)
+					}
+					logContent, err := ioutil.ReadAll(logFile)
+					if err != nil {
+						return errors.New("slirp4netns failed")
```
Comment (rhatdan) [inline libpod/networking_linux.go]: errors.Wrapf(err, "slirp4netns failed")

Comment (rhatdan) [reply libpod/networking_linux.go]: Or at least slirp4netns failed, unable to read output of slirp4netns command.

Comment (giuseppe) [reply libpod/networking_linux.go]: fixed

**Comment by giuseppe** [comment]
> Where do we close the FD for the log file?

Thanks for noticing it.  I've amended a change in the last version.

**Pull request merged**